File: test/test_ts/test_limits.ts
import { Payload, BufferWriter } from "./schema_v2.ts";

const p = new Payload();
p.uuid = "a".repeat(65536);

const writer = new BufferWriter();
try {
    p.serialize(writer);
    console.error("expected Error for string length > 65535");
    Deno.exit(1);
} catch (e) {
    if (!(e instanceof Error) || !e.message.includes("uint16")) {
        console.error("unexpected error:", e);
        Deno.exit(1);
    }
}

console.log("TypeScript limits test: PASSED");

File: test/test_py/test_limits.py
import sys
from schema_v2 import Payload

long_uuid = "a" * 65536
p = Payload()
p.uuid = long_uuid

buf = bytearray()
try:
    p.serialize(buf)
except ValueError as e:
    if "uint16" not in str(e):
        print(f"unexpected error: {e}")
        sys.exit(1)
    print("Python limits test: PASSED")
    sys.exit(0)

print("expected ValueError for string length > 65535")
sys.exit(1)